include/cuda/std/detail/libcxx/include/optional

```diff
@@ -236,8 +283,13 @@ struct __optional_destruct_base<_Tp, false>
         :  __val_(_CUDA_VSTD::forward<_Args>(__args)...),
            __engaged_(true) {}
 
+  template <class _Fp, class... _Args>
+  _LIBCUDACXX_INLINE_VISIBILITY
+  constexpr __optional_destruct_base(__optional_construct_from_invoke_tag, _Fp&& __f, _Args&&... __args)
+      : __val_(_CUDA_VSTD::invoke(_CUDA_VSTD::forward<_Fp>(__f), _CUDA_VSTD::forward<_Args>(__args)...)), __engaged_(true) {}
+
     _LIBCUDACXX_INLINE_VISIBILITY
-    void reset() noexcept
+    _LIBCUDACXX_CONSTEXPR_AFTER_CXX11 void reset() noexcept
```

Comment:
That should only be constexpr with C++20 due to the destructor call

Reply:
Disagree - if the type in the optional has a trivial destructor, this will work pre-20, no?

Reply:
It would, but that here is the specialization where `is_trivially_destructible` is false